src/stores/ui.ts

```diff
@@ -1,5 +1,19 @@
+import { computed, action, observable } from 'mobx';
 import { NavigateFn } from '@reach/router';
 
 export class UiStore {
+  @observable public isLoading = false;
+
+  @observable public errors: String[] = [];
```

Comment:
One of the things that will help with autocompletion is to use the MobX types on these observables.

```js
@observable public errors: IObservableArray<string> = [];

// or, you can do this...
public errors: IObservableArray<string> = observable.array([]);
```

We should probably pick one way or the other and do that all the time.

Back to the point... If you type your observables using the MobX types, you will notice that the observable arrays now have a `.replace` on them (and some other methods as well).

Reply:
In the first case, `IObservableArray` is needed to type `errors`. In the second case, it would not be needed since the array is wrapped in `observable.array([])`.